### PR TITLE
[Xcode12] Bring xharness changes to xcode12. 

### PR DESCRIPTION
**Pull request by mandel-macaque**
Bring all the changes that have happen in main on xharness to xcode12. In order to review this, you can focus on the diff between main and this PR using:

```bash
git diff main  -- tests/xharness
```

This will result in a much smaller diff than then one against xcode12. Diff can be found [here](https://gist.github.com/mandel-macaque/9486d72d5fce1908d8fa690152dfe83a).

Plans:

Since the history is in main, I plan to do a squash of this change, merge with xcode12.2. Once we merge xcode12.2 with main, revert the change and then merge to make the merge cleaner.

### CODE REVIEW

File: tests/bcl-test/Make.frag
Comment (rolfbjarne) [inline tests/bcl-test/Make.frag:1]: This file was removed in main (39629ab93b8d723b472a52378615dc5cf03553bd), so why is it added here?

File: tests/bcl-test/Makefile
Comment (rolfbjarne) [inline tests/bcl-test/Makefile:1]: Same: this file does not exist in main anymore.

File: tests/bcl-test/Make.frag
Comment (mandel-macaque) [reply tests/bcl-test/Make.frag:1]: Mistake :)

File: tests/bcl-test/Makefile
Comment (mandel-macaque) [reply tests/bcl-test/Makefile:1]: Same, a mistake, will remove

**Review by dalexsoto — APPROVED** [review]
Tests results look reasonable, I mean failing but due to issues to the tests and not xharness so 👍

**Review by tj-devel709 — APPROVED** [review]

**Review by rolfbjarne — COMMENTED** [review]
The changes look good, but it's a _very_ big change to a release branch, and contains shipping changes as well, so I'm not sure if this is the best way to do this. I'd feel much more comfortable with a targeted fix that only handles the issue we're trying to fix. OTOH there might be no problems... so I'll leave it to @chamons to decide :smile:

**Comment by mandel-macaque** [comment]
@rolfbjarne looking at the diff, we have the possible changes that might ship: 

* mk/xamarin.mk
* src/Makefile
* src/arkit.cs
* tools/common/.gitignore
* tools/common/Make.common
* tools/common/create-makefile-fragment.sh
* tools/mmp/mmp.csproj
* tools/mtouch/Assembly.cs
* tools/mtouch/mtouch.csproj

I'll try to revert those changes and if tests run, we will minimize the danger.

**Comment by mandel-macaque** [comment]
Do not merge, we will be using this PR as an integration branch to get device tests.

**Comment by mandel-macaque** [comment]
No longer required.

**Pull request closed without merging**
